### PR TITLE
feat: [kubelet/cAdvisor metrics package] add NodeMonitoring Status support

### PR DESCRIPTION
**Pull request by macxamin**

### CODE REVIEW

File: pkg/operator/collection.go [hunk 1]
```diff
@@ -503,6 +503,42 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
...
+		change, err := nm.Status.SetMonitoringCondition(nm.GetGeneration(), metav1.Now(), cond)
+		if err != nil {
+			// Log an error but let operator continue to avoid getting stuck
+			// on a potential bad resource.
+			logger.Error(err, "setting nodemonitoring status state")
+		}
+
+		if change {
+			// convert ScrapeNodeEndpoint to ScrapeEndpoint.
+			var scrapeEndpoints []monitoringv1.ScrapeEndpoint
```
Comment (macxamin) [inline pkg/operator/collection.go:516]: Do you have any better ideas on how to use NodeMonitoringCRD and PodMonitoringCRD in statusUpdates?

File: pkg/operator/apis/monitoring/v1/types_test.go [hunk 2]
```diff
@@ -1265,19 +1265,19 @@ func TestSetPodMonitoringCondition(t *testing.T) {
 		cond       *MonitoringCondition
 		generation int64
 		now        metav1.Time
-		curr, want *PodMonitoringStatus
+		curr, want *MonitoringStatus
```
Comment (TheSpiritXIII) [inline pkg/operator/apis/monitoring/v1/types_test.go:1268]: nit: maybe also rename function (remove the `Pod` prefix)?

File: pkg/operator/collection.go [hunk 3]
```diff
@@ -503,6 +503,42 @@ func (r *collectionReconciler) makeCollectorConfig(ctx context.Context, spec *mo
...
+				Spec: monitoringv1.PodMonitoringSpec{
+					Selector:  nm.Spec.Selector,
+					Endpoints: scrapeEndpoints,
+					Limits:    nm.Spec.Limits,
+				},
+				Status: monitoringv1.PodMonitoringStatus{
+					MonitoringStatus: nm.Status,
+				},
+			}
+			r.statusUpdates = append(r.statusUpdates, &nodeMonConv)
```
Comment (TheSpiritXIII) [inline pkg/operator/collection.go:540]: issue: I'm not sure if this is right. `statusUpdates` is for updating statuses of _existing_ [Cluster]PodMonitorings but one doesn't exist here. I believe your intention was to pass the NodeMonitoring object here.  An e2e test would have caught this.

I would rename `PodMonitoringCRD.GetStatus` to something like `PodMonitoringCRD.GetPodMonitoringStatus` and create a new interface with a method `MonitoringCRD.GetStatus` that returns a `MonitoringStatus`. Then, `statusUpdates` can store a `MonitoringCRD` and you'd be able to pass in your `NodeMonitoring` here.

**Review by TheSpiritXIII — CHANGES_REQUESTED** [review]
Just a few nits and an issue to look at. Great job otherwise!

**Comment by macxamin** [comment]
Closing, this is no longer relevant after the types.go split and https://github.com/GoogleCloudPlatform/prometheus-engine/pull/771.

**Pull request closed without merging**